modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/lowstorage/LowStorageDialog.kt

```diff
@@ -0,0 +1,169 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
+import au.com.shiftyjelly.pocketcasts.compose.bottomsheet.ModalBottomSheet
+import au.com.shiftyjelly.pocketcasts.compose.buttons.RowButton
+import au.com.shiftyjelly.pocketcasts.compose.buttons.RowOutlinedButton
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH20
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH50
+import au.com.shiftyjelly.pocketcasts.compose.preview.ThemePreviewParameterProvider
+import au.com.shiftyjelly.pocketcasts.compose.theme
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import au.com.shiftyjelly.pocketcasts.utils.Util
+import kotlinx.coroutines.launch
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun LowStorageLaunchBottomSheet(
+    parent: ViewGroup,
+    modifier: Modifier = Modifier,
+    shouldShow: Boolean = true,
+    totalDownloadSize: Long,
+    onManageDownloadsClick: () -> Unit,
+    onMaybeLaterClick: () -> Unit,
+    onExpanded: () -> Unit,
+) {
+    val sheetState = rememberModalBottomSheetState(
+        initialValue = ModalBottomSheetValue.Hidden,
+        skipHalfExpanded = true,
+    )
+    val coroutineScope = rememberCoroutineScope()
+
+    ModalBottomSheet(
+        parent = parent,
+        shouldShow = shouldShow,
+        customSheetState = sheetState,
+        onExpanded = onExpanded,
+        customContent = {
+            LowStorageDialog(
+                modifier = modifier,
+                totalDownloadSize = totalDownloadSize,
+                onManageDownloadsClick = {
+                    coroutineScope.launch { sheetState.hide() }
+                    onManageDownloadsClick.invoke()
+                },
+                onMaybeLaterClick = {
+                    coroutineScope.launch { sheetState.hide() }
+                    onMaybeLaterClick.invoke()
+                },
+            )
+        },
+    )
+}
+
+@Composable
+internal fun LowStorageDialog(
+    modifier: Modifier = Modifier,
+    totalDownloadSize: Long,
+    onManageDownloadsClick: () -> Unit,
+    onMaybeLaterClick: () -> Unit,
+) {
+    val formattedTotalDownloadSize = Util.formattedBytes(bytes = totalDownloadSize, context = LocalContext.current)
+
+    Box(
+        modifier = modifier
+            .fillMaxWidth(),
+        contentAlignment = Alignment.Center,
+    ) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = modifier.fillMaxWidth(),
+        ) {
```

Comment:
```suggestion
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = modifier.fillMaxWidth().padding(horizontal = 21.dp),
    ) {
```
Does there need to be the parent of `Box`?
Instead of adding the horizontal padding to the child components, could you add it here instead?

Reply:
I have simplified the layout